pkg/mdformatter/mdgen/mdgen.go

```diff
@@ -100,6 +111,53 @@ func (t *genCodeBlockTransformer) TransformCodeBlock(ctx mdformatter.SourceConte
 			return nil, errors.Errorf("expected language a first element of info string got %q; Got info string %q", lang, string(infoString))
 		}
 	}
+
+	// Code fence with file.
+	fileToCopy, fileOk := infoStringAttr[infoStringKeyFile]
+	linesToCopy, lineOk := infoStringAttr[infoStringKeyLines]
+	if fileOk {
+		file, err := os.Open(fileToCopy)
+		if err != nil {
+			return nil, errors.Errorf("file could not be opened, ensure file path: %q is correct. Got info string %q", infoStringKeyFile, string(infoString))
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
+		scanner.Split(bufio.ScanLines)
+		var text []byte
+
+		switch lineOk {
+		case true:
+			val := strings.Split(linesToCopy, "-")
+			start, errStart := strconv.Atoi(val[0])
+			if errStart != nil {
+				return nil, errors.Errorf("start line value isn't a number: %q. Got info string %q", val[0], string(infoString))
+			}
+			end, errEnd := strconv.Atoi(val[1])
+			if errEnd != nil {
+				return nil, errors.Errorf("end line value isn't a number: %q. Got info string %q", val[1], string(infoString))
+			}
+			if start >= end {
+				return nil, errors.Errorf("line number range isn't valid: %q-%q. Got info string %q", val[0], val[1], string(infoString))
+			}
+			line := 0
+
+			for scanner.Scan() {
+				if line >= start && line <= end {
+					text = append(text, scanner.Bytes()...)
+					text = append(text, "\n"...)
+				}
+				line++
+			}
+		case false:
```

Comment:
Can we reuse the above, but just use start 0 and end = `math.MaxInt64`? (:

Reply:
Sure. Will make the necessary changes!